File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def traverse(currentEl, index, ans):
            if index==len(nums):
                copy = currentEl.copy()
                ans.append(copy)
                return
            
            currentEl.append(nums[index])
            traverse(currentEl, index+1, ans)
            currentEl.pop()
            traverse(currentEl, index+1, ans)
        
        traverse([], 0, res)
        return res

File: Data Structures & Algorithms/remove-element/submission-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int writeIndex = 0;

        for(int i=0; i<nums.length; i++){
            if(nums[i]!=val){
                nums[writeIndex]=nums[i];
                writeIndex+=1;
            }
        }

        return writeIndex;
    }
}